src/Microsoft.AspNetCore.Rewrite/Pattern.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Rewrite
+{
+    /// <summary>
+    /// Contains a sequence of pattern segments, which on obtaining the context, will create the appropriate
+    /// test string and condition for rules and conditions.
+    /// </summary>
+    public class Pattern
+    {
+        private IEnumerable<PatternSegment> Segments { get; }
```

Comment:
Why is this `IEnumerable<PatternSegment>` instead of IList<PatternSegment>` or `IReadOnlyList<PatternSegment>`